cares_reinforcement_learning/memory/memory_buffer.py

```diff
@@ -1,201 +1,23 @@
-from __future__ import annotations
-
+import random
 from collections import deque
 
-import numpy as np
-
-# pylint: disable-next=wildcard-import, unused-wildcard-import
-from cares_reinforcement_learning.memory.augments import *
-
 
 class MemoryBuffer:
-    """
-    This class represents a base memory buffer used in Reinforcement Learning (RL).
-    The buffer is used to store experiences of the interactions of an agent with its environment.
-
-    The buffer has a maximum capacity and uses a deque to store the experiences,
-    because it is efficient for appending and popping on both ends especially when
-    the maximum length is reached.
-
-    Attributes
-    ----------
-    max_capacity : int
-        The maximum capacity of the buffer.
-    buffers : dict
-        A dictionary to hold different buffers for easy management.
-    """
-
-    def __init__(
-        self, max_capacity: int = int(1e6), eps=1e-6, alpha=0.6, augment=std, **params
-    ):
-        """
-        The constructor for MemoryBuffer class.
-
-        Parameters
-        ----------
-        max_capacity : int
-            The maximum capacity of the buffer (default is 1,000,000).
-        dtype : np value type
-            Sets the value type data is stored as
-        """
-        self.max_capacity = max_capacity
-        self.head = 0
-        self.full = False
-
-        self.buffers = {"priorities": deque(maxlen=max_capacity)}
-
-        self.augment = augment
-        self.params = {"eps": eps, "alpha": alpha}
-        for key, value in params.items():
-            self.params[key] = value
-
-    def add(self, **experience):
-        """
-        Adds an experience to the memory buffer. Each key-value pair in the experience dictionary is added
-        to a corresponding buffer. If a buffer for a given key does not exist, it is created.
-
-        If the memory buffer is full, this function will overwrite the oldest experience with the new one.
-
-        Parameters
-        ----------
-
-        **experience : dict
-            A dictionary where the key is the name of the experience and the value is the experience data.
-            The data can be of any type.
-
-        Examples
-        --------
-        >>> memory.add([1,2,3], [2,3,4], action=[1.5, 2.1],
-        >>>            reward=-0.32, done=False)
-        """
-        for key, value in experience.items():
-            if key not in self.buffers:
-                self.buffers[key] = [None] * self.max_capacity
-
-            self.buffers[key][self.head] = value
-
-        if self.buffers.get("priorities") is None:
-            self.buffers["priorities"] = deque(maxlen=self.max_capacity)
-        max_priority = (
-            max(self.buffers["priorities"]) if self.buffers["priorities"] else 1.0
```

Comment:
This probably contributes to long run time. This is O(n) on every addition which is not great when we step every fraction of a second.